### ISSUE TITLE
models created by YOLO(..) can not be set to eval mode directly

### ISSUE THREAD

**Issue by WYYAHYT** [ultralytics/ultralytics#17751]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

The `YOLO` class is a subclass of `ultralytics.engine.Model`, and `ultralytics.engine.Model` is a subclass of `nn.Module`, so the model created by `YOLO`, for instance, `model = YOLO("yolo11n.pt")` can be set to `eval mode` logically.

However, when excute `model.eval()`, it launches training progress unexpectly, since the `YOLO` class overwrites `train()` function.

In concrete terms, `model.eval()` first called `eval()` function built in `nn.Module`, and here is the code:
```python
# In nn.Module
def eval(self: T) -> T:
        r"""Sets the module in evaluation mode.

        This has any effect only on certain modules. See documentations of
        particular modules for details of their behaviors in training/evaluation
        mode, if they are affected, e.g. :class:`Dropout`, :class:`BatchNorm`,
        etc.

        This is equivalent with :meth:`self.train(False) <torch.nn.Module.train>`.

        See :ref:`locally-disable-grad-doc` for a comparison between
        `.eval()` and several similar mechanisms that may be confused with it.

        Returns:
            Module: self
        """
        return self.train(False)
```
It can be seen that `eval()` return `self.train(False)`, this line exactly aims to call the `train()` function defined in `nn.Module`, as for now, `YOLO` inherits `nn.Module` and overwrites `train()` function,  so the `eval()` function actually calls `YOLO: train()`, instead of `nn.Module: train()`, 

And the `YOLO: train()` is designed to train the model, so the operation of calling `model.eval()`, triggers  **training progress.**

### Environment

Ultralytics 8.3.29 🚀 Python-3.8.12 torch-1.11.0 CUDA:0 (NVIDIA GeForce RTX 4090, 24217MiB)
Setup complete ✅ (128 CPUs, 1007.5 GB RAM, 2085.6/28498.9 GB disk)

OS                  Linux-5.4.0-200-generic-x86_64-with-glibc2.17
Environment         Docker
Python              3.8.12
Install             git
RAM                 1007.53 GB
Disk                2085.6/28498.9 GB
CPU                 Intel Xeon Platinum 8369B 2.90GHz
CPU count           128
GPU                 NVIDIA GeForce RTX 4090, 24217MiB
GPU count           2
CUDA                11.3

numpy                    ✅ 1.24.4>=1.23.0
matplotlib               ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow                      ✅ 9.0.1>=7.1.2
pyyaml                    ✅ 6.0>=5.3.1
requests                  ✅ 2.28.2>=2.23.0
scipy                        ✅ 1.10.1>=1.4.1
torch                       ✅ 1.11.0>=1.8.0
torchvision              ✅ 0.12.0>=0.9.0
tqdm                       ✅ 4.65.2>=4.64.0
psutil                       ✅ 5.8.0
py-cpuinfo              ✅ 9.0.0
pandas                    ✅ 2.0.3>=1.1.4
seaborn                   ✅ 0.13.2>=0.11.0
ultralytics-thop       ✅ 2.0.11>=2.0.0
numpy                    ✅ 1.24.4<2.0.0; sys_platform == "darwin"
torch                       ✅ 1.11.0!=2.4.0,>=1.8.0; sys_platform == "win32"

### Minimal Reproducible Example

```python
from ultralytics import YOLO

# Create minimal reproducible example
model = YOLO('yolov8n.pt')

print(f"Initial training state: {model.model.training}")

# Test eval()
model.eval()  # ---> here, triggers training progress
print(f"After eval(): {model.model.training}")
```

### Additional

Although `model.model.eval()` is useful, but maybe it is not elegant enough......

Based on the design of wrapper, `YOLO` has an attrbute named `model`, actually, what I want is setting this attribute to`eval mode`, so, we can add a function called `eval() `in `Model` class (the class that `YOLO` inherits from), so as to avoid calling `nn.Module: eval()`, and here is the code:
```python
class Model:
    ...
    def eval(self):
        self.model.eval()
        return self
```

 

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WYYAHYT, thank you for your detailed report on the `YOLO` class behavior 🚀! This is an automated response to guide you while we have an Ultralytics engineer look into the details of your issue.

To address your concerns about using `eval()` on a `YOLO` object inadvertently triggering training due to method overriding, we appreciate your proposed solution and willingness to contribute a PR! Meanwhile, please ensure your setup is aligned with the latest versions, as improvements might already exist.

If you haven't upgraded recently, here's how you can upgrade to the latest `ultralytics` package, ensuring all requirements are met within a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

For any reproducible issue 🐛, providing further context like a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) is immensely helpful.

For additional help and community support, consider joining our [Discord](https://ultralytics.com/discord) 🎧 for real-time chat, or engage with others on [Discourse](https://community.ultralytics.com) and our [Subreddit](https://reddit.com/r/ultralytics).

We appreciate your engagement and look forward to seeing your contribution to our codebase! 🤝

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, it indicates all our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing, confirming the stability across macOS, Windows, and Ubuntu.

**Comment by Laughing-q** [comment]
@WYYAHYT Thanks for reporting! I've opened a PR to fix this! #17754

**Comment by WYYAHYT** [comment]
@Laughing-q Thanks a lot!

**Comment by WYYAHYT** [comment]
@Laughing-q I have wrote the code of accessing model.training, how can I open a PR based on your PR that has not been merged yet? Since it is based on your PR. Should I wait util your PR is merged?

**Comment by Laughing-q** [comment]
@WYYAHYT yes you could directly open a PR based on my branch, no worries. And if there's conflict we could easily fix it.
